modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1564,18 +1567,17 @@ def cat_codes(self):
         assert len(self.columns) == 1
         assert self._dtypes[-1] == "category"
 
-        col = self.columns[-1]
         exprs = self._index_exprs()
-        col_expr = self.ref(col)
+        col_expr = self.ref(self.columns[-1])
         code_expr = OpExpr("KEY_FOR_STRING", [col_expr], get_dtype("int32"))
         null_val = LiteralExpr(np.int32(-1))
-        exprs[col] = build_if_then_else(
+        exprs[None] = build_if_then_else(
```

Comment:
Why?

Reply:
right, shouldn't it be something like:
```suggestion
        exprs[MODIN_UNNAMED_SERIES_LABEL] = build_if_then_else(
```